### PR TITLE
Provide error messages for assertion failures

### PR DESCRIPTION
**Pull request by TomJKing**
Error message to help when trying diagnose why a test has failed

In some case a simple assertion error in the logs does not make it clear what exactly has failed

### CODE REVIEW

File: src/test/scala/helpers/logging/AssertionErrorMessages.scala [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package helpers.logging
+
+object AssertionErrorMessages {
+  def elementMissingMessage(selector: String): String = s"'$selector' element missing"
```
Comment (suzannehamilton) [inline src/test/scala/helpers/logging/AssertionErrorMessages.scala:4]: This is definitely better than what we had before, so I'm happy to approve it as it is.

I have a suggestion to make it even clearer, but feel to tell me you've done as much as you want to do at the moment!

The suggestion is to wrap the assertions, to give them better names:

Currently the assertions look like this:

```
    val errorElements: util.List[WebElement] = webDriver.findElements(By.cssSelector(errorClassName))
    Assert.assertNotNull(errorElements)
```

I think the tests would be even clearer if we could say something like:

```
  assertElementWithClassExists(errorClassName);
```

Where `assertElementWithClassExists` is a helper method that does the assertion, including the new failure message you've just added.

That's why I was asking about Selenium's in-built assertions on Slack - this kind of thing would be so useful that I'm surprised they don't already exist.

As I said, this is just a thought - I'm happy to merge this as it is and revisit custom assertions later.

File: src/test/scala/steps/Steps.scala [hunk 2]
```diff
@@ -205,23 +205,27 @@ class Steps extends ScalaDsl with EN with Matchers {
...
 
-      Assert.assertTrue(errorElement.getText.contains(errorMessage))
+      val actualErrorText = errorElement.getText
+      Assert.assertTrue(doesNotContain(actualErrorText, errorMessage), actualErrorText.contains(errorMessage))
   }
 
   And("^the user will see the error message (.*)") {
     errorMessage: String =>
+      val selector = ".govuk-error-summary__list"
       val errorElement = webDriver.findElement(By.cssSelector(".govuk-error-summary__list"))
```
Comment (suzannehamilton) [inline src/test/scala/steps/Steps.scala:219]: Did you mean to make this use the `selector` variable too?

Comment (TomJKing) [reply src/test/scala/steps/Steps.scala:219]: Yes. Thanks for spotting this.

File: src/test/scala/helpers/logging/AssertionErrorMessages.scala (hunk 1 above)
Comment (TomJKing) [reply src/test/scala/helpers/logging/AssertionErrorMessages.scala:4]: I'm thinking along the same lines. I'm a bit surprised that Selenium doesn't do this. I'm sure they did from working with it previously, although a long time ago. Will spend a bit more time looking at this later.

**Review by suzannehamilton — APPROVED** [review]

**Pull request merged**